runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -962,45 +1220,422 @@ TR_VectorAPIExpansion::validateVectorAliasClasses(TR_BitVector * vectorAliasTabl
+
+   treeTop->insertBefore(TR::TreeTop::create(comp(), TR::Node::create(TR::treetop, 1, newArray)));
+
+   // Generate vector store to the payload array
+
+   TR::VectorLength vectorLength = OMR::DataType::bitsToVectorLength(bitsLength);
+   TR::DataType opCodeType = (objectType == Vector) ?
+                               TR::DataType::createVectorType(elementType, vectorLength)
+                               : TR::DataType::createMaskType(elementType, vectorLength);
+
+   TR::Node *vloadNode = child;
+
+   if (!child->getOpCode().isVectorOpCode())  // not vectorized yet
+      vloadNode = vectorizeLoadOrStore(this, child, opCodeType, true);
+
+   TR::Node *aladdNode = generateArrayElementAddressNode(comp(), newArray, TR::Node::lconst(node, 0), elementSize);
+
+   TR::SymbolReference *vectorShadow = comp()->getSymRefTab()->findOrCreateArrayShadowSymbolRef(opCodeType, NULL);
+   TR::ILOpCodes storeOpcode = TR::ILOpCode::createVectorOpCode(TR::vstorei, opCodeType);
+   TR::Node *storeNode = TR::Node::createWithSymRef(storeOpcode, 2, aladdNode, vloadNode, 0, vectorShadow);
+   treeTop->insertBefore(TR::TreeTop::create(comp(), storeNode));
+   TR::Node *fence = TR::Node::createAllocationFence(newArray, newArray);
+   //fence->setAllocation(NULL);
+   treeTop->insertBefore(TR::TreeTop::create(comp(), fence));
+
+   // generate "new  jitNewObject"
+   TR::Node *newObject = TR::Node::create(child, TR::New, 1);
+   newObject->setSymbolReference(comp()->getSymRefTab()->findOrCreateNewObjectSymbolRef(comp()->getMethodSymbol()));
+
+   TR_J9VMBase *fej9 = comp()->fej9();
+   TR::VMAccessCriticalSection getClassFromSignature(fej9);
+   TR::SymbolReference *j9class = comp()->getSymRefTab()->findOrCreateClassSymbol(comp()->getMethodSymbol(), -1, vecClass);
+
+   TR_ASSERT_FATAL(j9class, "J9Class symbol reference should not be null");
+
+   newObject->setAndIncChild(0, TR::Node::createWithSymRef(child, TR::loadaddr, 0, j9class));
+   treeTop->insertBefore(TR::TreeTop::create(comp(), TR::Node::create(TR::treetop, 1, newObject)));
+
+   // anchor old child
+   treeTop->insertBefore(TR::TreeTop::create(comp(), TR::Node::create(TR::treetop, 1, child)));
+   child->recursivelyDecReferenceCount();
+
+   node->setAndIncChild(i, newObject);
+
+   fence = TR::Node::createAllocationFence(newObject, newObject);
+   //fence->setAllocation(NULL);
+   treeTop->insertBefore(TR::TreeTop::create(comp(), fence));
+
+   TR::SymbolReference *payloadSymRef = createPayloadSymbolReference(comp(), vecClass);
+   treeTop->insertBefore(TR::TreeTop::create(comp(), TR::Node::storeToAddressField(comp(), newObject, payloadSymRef, newArray)));
+
+   fence = TR::Node::createAllocationFence(newObject, newObject);
+   //fence->setAllocation(NULL);
+   treeTop->insertBefore(TR::TreeTop::create(comp(), fence));
+
+   if (_trace)
+      traceMsg(comp(), "Boxed: %dth child of node %p into %p\n", i, node, newObject);
+
+   if (TR::Options::getVerboseOption(TR_VerboseVectorAPI))
```

Comment:
I would prefer the optimizer to use compilation logs instead of vlog. Vlog is typically used for runtime. Verbose options are global and they are not passed to the JITServer If the method is compiled remotely, these messages will not appear.